### DISCUSSION TITLE
Wemos OLED Lolin32 porting

### DISCUSSION THREAD

**Discussion by costa-victor** [Moddable-OpenSource/moddable#863] [Q&A] [answered]
Hello there, I would like to understand the Moddable's plan before create a PR for porting a new development board.

I'm currently using the [Wemos OLED Lolin32](https://randomnerdtutorials.com/esp32-built-in-oled-ssd1306/) board, which is a cheap ESP32 OLED board I boughted a few months ago, showed in the image below:
![wemos-lolin-esp32-built-in-oled](https://user-images.githubusercontent.com/35344574/156272295-2ef580f6-7a4c-4a46-b018-dc926e98b9b4.jpg)

Regarding this port, I have two questions:

1. Is there a list of licensed boards on Moddable? Or all boards port welcome?
2. I couldn't find vendor references, however it's a popular board due to its low price. Is that a problem?

Thanks,
Victor

**Comment by phoddie** [comment, marked as answer]
Support for new device targets is welcome. Your board looks similar to the [Heltec WiFi Kit32](https://heltec.org/project/wifi-kit-32/). You might want to take a look at [the work](https://github.com/Moddable-OpenSource/moddable/tree/public/build/devices/esp32/targets/heltec_wifi_kit_32) @tribalmatt did to support that.

**Comment by tribalmatt** [comment]
I would expect that to be very straightforward.  If you can point out the
boards pin outs I can probably show you the changes to make.

**Comment by tribalmatt** [comment]
In the heltec manifest.json file.  If you update the sda and scl pins to
match your board...I would expect it to work.  The heltec requires the
reset pin to toggle to turn on the screen.  But I don't see on your boards
pinouts.

"sda_pin": 4,

"scl_pin": 15,


Should become

"sda_pin": 5,

"scl_pin": 4,


Good luck

**Reply by costa-victor** [reply]
Thank you @tribalmatt , I appreciate this. As you said, it was very straightforward due to your previous job and also because the boards are quite similar.

It works as expected and you can find it [here](https://github.com/Moddable-OpenSource/moddable/pull/864).

Thank you again and kudos for your great job there.
